src/index.ts

```diff
@@ -523,6 +604,36 @@ function canDisplayGraphiQL(request: Request, params: GraphQLParams): boolean {
   return !params.raw && accepts(request).types(['json', 'html']) === 'html';
 }
 
+/**
+ * Helper function for sending part of a multi-part response using only the core Node server APIs.
+ */
+function sendPartialResponse(
+  pretty: boolean,
+  response: Response,
+  result: FormattedExecutionResult | FormattedExecutionPatchResult,
+): void {
+  const json = JSON.stringify(result, null, pretty ? 2 : 0);
+  const chunk = Buffer.from(json, 'utf8');
```

Comment:
will this always be utf8?

Reply:
good question, I'm fairly sure that `utf8` is the only way that `application/json` content can be encoded

Reply:
this has me wanting to bring up protocolbuffers, messagepack and other binary formats, as well as yaml, however that seems way out of scope for this PR, and should probably begin as an HTTP transport RFC proposal. GraphiQL users have on multiple occasions requested support for these formats, especially for server-to-server transmissions and iot

back to this discussion, though, would it work to use a binary encoding format `latin1` with JSON? I've never tried that myself. if someone was returning a binary format like BSON for example, then we should be returning content type of `application/bson`, right?  Is there a case where someone would be sending `application/json`?

if there are indeed more ways to mix and match charset and content type with JSON than I'm aware of, maybe this should be controlled by the request headers then, and fall back to `utf8` as default?

Reply:
Worth noting that the transport RFC spec doesn't require `utf8` necessarily (yet):
https://github.com/graphql/graphql-over-http/blob/master/rfcs/IncrementalDelivery.md

and the reference client implementation expects `utf8`:
https://github.com/relay-tools/fetch-multipart-graphql/blob/dfa54d0c649ecbbedac3e9764998f63894367505/src/fetch.js#L26

Reply:
meros always at this time uses UTF8. It's a force of habit. But there might be people out there needing some out of the ordinary encodings. 

Maybe it's a cross that bridge when we get there?

Reply:
express-graphql already only sends utf8 so I don't think there's any change in behavior here: https://github.com/graphql/express-graphql/blob/master/src/index.ts#L531

As far as the spec, I'm not sure if it's worth specifying what encodings are required. FWIW it seems like JSON requires either UTF-8, UTF-16, or UTF-32 but recommends UTF8 for maximum interoperablity.

https://tools.ietf.org/html/rfc7159#section-8.1